### DISCUSSION TITLE
How to troubleshoot strange rust-analyzer behavior

### DISCUSSION THREAD

**Discussion by vsiles** [mrcjkb/rustaceanvim#464] [Q&A]
I have a strange behavior where rust-analyzer no longer shows diagnostics in nvim. I identified one occurrence (and can't explain why :D) but it's not the only one.
Here's a repro:

```
$ cargo new test-project
$ cd test-project
$ cargo add whoami
$ nvim src/main.rs # add a line `const X : bool = 42;`, it should show an error
$ echo "fn main() { let _ = whoami::username(); }" > build.rs
$ nvim src/main.rs # no more error !
$ nvim -u minimal.lua src/main.rs # no error either
```

If I run `rust-analyzer diagnostics .` in the console, or `cargo check`, I do get the error. But with that simple `build.rs`, I don't have them in the editor.

I have other projects where I can't pinpoint what is bothering RA. How can I debug this ? I tried to set `RA_LOG=info` or `debug` but I don't see anything relevant. Any suggestions ?

PS: I tried opening the file with `helix` to make sure it was something to do with the editor, not rust-analyzer, and I see the error with or without the build file. That's why I think something's fishy around nvim/rustaceanvim

**Comment by mrcjkb** [comment]
Hey 👋 

This is not enough information for me to help you. 
Please see the troubleshooting section in the readme and the bug report template.

**Reply by vsiles** [reply]
Thank you for the fast feedback. Before I bother you more here, how would you configure nvim/rustaceanvim/RA to get more information/log ? I think the issue is that RA fails to build the project, so can't give meaningful type errors.
However when I look in the lsp logs (in nvim, `:LspLog`), it's virtually empty. When I look in RA's log (in nvim, `:RustLsp logFile`), it is totally empty.

Could you share a minimal configuration to try and get more logs out of my setup ?

**Reply by mrcjkb** [reply]
You might have to play around with environment variables to configure log levels, etc.

See also: https://github.com/rust-lang/rust-analyzer/blob/master/docs/dev/README.md#logging
